src/interface/update.hpp

```diff
@@ -66,61 +66,31 @@ void AverageIndependentData(T &c1, T &c2, const Real wgt1) {
       });
 }
 
-template <typename T>
-bool IsBlock(std::shared_ptr<T> &rc) {
-  std::string mytype = typeid(rc).name();
-  return (mytype.find("MeshBlockData") != std::string::npos);
-}
-
-template <typename T>
-using TimeStepFuncType = Real(std::shared_ptr<T> &);
-
 template <typename T>
 TaskStatus EstimateTimestep(std::shared_ptr<T> &rc) {
-  using Tstep_t = TimeStepFuncType<T>;
   using Desc_t = std::shared_ptr<StateDescriptor>;
-  auto pm = rc->GetGridPointer();
   Real dt_min = std::numeric_limits<Real>::max();
-  std::string fname = (IsBlock(rc) ? magic::estimate_dt_block : magic::estimate_dt_mesh);
+  auto pm = rc->GetGridPointer();
   for (const std::pair<std::string, Desc_t> &pkg : pm->packages) {
```

Comment:
could this be `for(const auto & pkg : pm->packages)`? You might be able to remove the `Desc_t` using statement then too